solidity/test/Oracle_test.js

```diff
@@ -83,28 +84,60 @@ contract('Oracle', () => {
     context("when called through the LINK token", () => {
       let log, tx;
       beforeEach(async () => {
-        let args = requestDataBytes(jobId, to, fHash, "id", "");
+        let args = requestDataBytes(specId, to, fHash, "id", "");
         tx = await requestDataFrom(oc, link, 0, args);
         assert.equal(3, tx.receipt.logs.length)
 
         log = tx.receipt.logs[2];
       });
 
       it("logs an event", async () => {
-        assert.equal(jobId, web3.toUtf8(log.topics[2]));
+        assert.equal(specId, web3.toUtf8(log.topics[2]));
       });
 
       it("uses the expected event signature", async () => {
-        // If updating this test, be sure to update TestServices_RunLogTopic_ExpectedEventSignature.
+        // If updating this test, be sure to update services.RunLogTopic.
         let eventSignature = "0x3fab86a1207bdcfe3976d0d9df25f263d45ae8d381a60960559771a2b223974d";
         assert.equal(eventSignature, log.topics[0]);
       });
     });
 
     context("when not called through the LINK token", () => {
+      it("reverts", async () => {
+        await assertActionThrows(async () => {
+          await oc.requestData(1, specId, to, fHash, "id", "", {from: oracleNode});
+        });
+      });
+    });
+  });
+
+  describe("#specAndRun", () => {
+    context("when called through the LINK token", () => {
+      let log, tx, amount;
+      beforeEach(async () => {
+        amount = 1337;
+        let args = specAndRunBytes(to, fHash, "requestId", "");
+        tx = await link.transferAndCall(oc.address, amount, args);
+        assert.equal(3, tx.receipt.logs.length)
+
+        log = tx.receipt.logs[2];
+      });
+
       it("logs an event", async () => {
+        assert.equal(amount, web3.toBigNumber(log.topics[2]));
+      });
+
+      it("uses the expected event signature", async () => {
```

Comment:
Can this test can ✂️ `async` or does it need to stay because of the `beforeEach` stuff?